### PR TITLE
fix(onboard step): [MAG-3448] add missing surface props to step component

### PR DESCRIPTION
**Pull request by eTheMan80**
# <JIRA issue / Title>

## TL;DR

-   Add missing surface props to Step component
-  
-  
-

## Details

Checklist / Reminders:

-   [x] Documentation / Story is created / adapted
-   [ ] Unit tests
-   [ ] CSS changes are BEM-compliant

Screenshot 2021-02-10 at 19 55 36
Screenshot 2021-02-10 at 19 45 55
Screenshot 2021-02-10 at 19 45 33

### CODE REVIEW

File: src/components/Molecules/Steps/Steps.tsx [hunk 1]
```diff
@@ -72,6 +73,7 @@ export const Steps: FC<StepsProps> = props => {
       elevation={elevation}
       noBorder={['top', 'right', 'bottom', 'left']}
       padding="md"
+      surface={surface}
```
Comment (tricaalexandru) [inline src/components/Molecules/Steps/Steps.tsx:76]: @matthill8286 is this what you were talking about?

**Review by tricaalexandru — APPROVED** [review]

File: src/components/Molecules/Steps/Steps.interface.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { ReactElement } from 'react'
 import { Elevation, ThemeColors } from '@/types/theme'
 
 export type StepsVariant = 'prominent' | 'simple' | 'uniform'
+export type StepsBackgroundColor = 'white' | 'grey1' | 'grey4' | 'black' | 'primary' | 'selected'
```
Comment (matthill8286) [inline src/components/Molecules/Steps/Steps.interface.ts:5]: @eTheMan80 When will the background ever be ` 'black' | 'primary' | 'selected'`

File: src/components/Molecules/Steps/Steps.tsx (hunk 1 above)
Comment (matthill8286) [reply src/components/Molecules/Steps/Steps.tsx:76]: Kind of the easiest i.e less code solution would be to revert the last 2 merged on these components and update this.

Screenshot 2021-02-11 at 11 01 05

So if no surface is passed into it, it'll default to transparent.

File: src/components/Molecules/Steps/Steps.interface.ts (hunk 2 above)
Comment (matthill8286) [reply src/components/Molecules/Steps/Steps.interface.ts:5]: @eTheMan80 This isn't needed, you can just do this in `src/components/Atoms/Card/Card.styled.ts`:

Screenshot 2021-02-11 at 11 01 05

File: src/components/Molecules/Steps/Steps.interface.ts [hunk 3]
```diff
@@ -62,6 +63,7 @@ export interface StepsProps {
   completedStep: number
   elevation?: Elevation
   color?: ThemeColors
+  surface?: StepsBackgroundColor
```
Comment (matthill8286) [inline src/components/Molecules/Steps/Steps.interface.ts:66]: doesn't need a surface

File: src/components/Molecules/Steps/Steps.tsx [hunk 4]
```diff
@@ -20,6 +20,7 @@ export const Steps: FC<StepsProps> = props => {
     color = 'primary',
     variant = 'prominent',
     onChange,
+    surface = 'white',
```
Comment (matthill8286) [inline src/components/Molecules/Steps/Steps.tsx:23]: not needed

File: src/components/Organisms/TopTeaser/TopTeaser.story.tsx [hunk 5]
```diff
@@ -26,7 +26,8 @@ const colorTypes: {
   selected: 'selected',
   white: 'white',
   black: 'black',
-  grey: 'grey4',
+  grey1: 'grey1',
+  grey4: 'grey4',
```
Comment (matthill8286) [inline src/components/Organisms/TopTeaser/TopTeaser.story.tsx:30]: not needed.

**Review by matthill8286 — COMMENTED** [review]
There is a much simpler solution to the background color of a step, i've commented below

File: src/components/Molecules/Steps/Steps.interface.ts (hunk 2 above)
Comment (matthill8286) [reply src/components/Molecules/Steps/Steps.interface.ts]: remove the default param from the Card and voila, 

Screenshot 2021-02-11 at 11 11 08
Screenshot 2021-02-11 at 11 10 41

Comment (eTheMan80) [reply src/components/Molecules/Steps/Steps.interface.ts]: @matthill8286 thanks, updating now 🙂

File: src/components/Molecules/Steps/Steps.tsx (hunk 1 above)
Comment (eTheMan80) [reply src/components/Molecules/Steps/Steps.tsx]: @matthill8286 thanks, this has been updated now 🙂

File: src/components/Molecules/Steps/Steps.interface.ts (hunk 3 above)
Comment (eTheMan80) [reply src/components/Molecules/Steps/Steps.interface.ts]: @matthill8286 thanks, this has been updated

File: src/components/Molecules/Steps/Steps.tsx (hunk 4 above)
Comment (eTheMan80) [reply src/components/Molecules/Steps/Steps.tsx]: @matthill8286 thanks, this has been updated

File: src/components/Organisms/TopTeaser/TopTeaser.story.tsx (hunk 5 above)
Comment (eTheMan80) [reply src/components/Organisms/TopTeaser/TopTeaser.story.tsx]: @matthill8286 thanks, this has been updated

**Review by matthill8286 — APPROVED** [review]
Perfect.

**Pull request merged**